src/main/java/frc/robot/hardware/rev/request/SparkMaxRequest.java

```diff
@@ -42,8 +44,11 @@ public T getSetPoint() {
 		return setPoint;
 	}
 
-	public Double getSetPointAsDouble() {
-		return setPointToDoubleConverter.apply(setPoint);
+	public Double getSetPointForSparkMAX() {
+		return controlType == CANSparkBase.ControlType.kVelocity
+			? setPointToDoubleConverter
+				.apply((T) Rotation2d.fromRotations(Conversions.perSecondToPerMinute(((Rotation2d) setPoint).getRotations())))
```

Comment:
split it to 2 lines, one for the unit conversions one for the toDouble conversion